gyp/pylib/gyp/xcode_emulation.py

```diff
@@ -1277,7 +1277,7 @@ def XcodeVersion():
   except:
     version = CLTVersion()
     if version:
-      version = re.match(r'(\d+\.\d+\.?\d*)', version).groups()[0]
+      version = re.match(r'(\d?\d\.\d?\d\.?\d?\d*)', version).groups()[0]
     else:
       raise GypError("No Xcode or CLT version detected!")
```

Comment:
On exception, let's add the version that was found...
` raise GypError("No Xcode or CLT version detected!  {}".format(version)`